ScrollingTreeScrollingNode: null check scrollingTree() in some places

tuankiet65 opened the pull request:
#### 52fb9ce2ac5dc83506f89a035029c51a3ff3c4e4
<pre>
ScrollingTreeScrollingNode: null check scrollingTree() in some places
<a href="https://rdar.apple.com/114615469">rdar://114615469</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=298686">https://bugs.webkit.org/show_bug.cgi?id=298686</a>

Reviewed by Simon Fraser.

AppKit might call into an orphan ScrollingTreeScrollingNode (node without
a scrolling tree) after the tree is torn down. As ScrollingTreeScrollingNode
holds a weak pointer to the owning scrolling tree, accessing the now-destroyed
tree crashes with a null pointer dereference. For now, null-check scrollingTree()
in places where the caller is from &quot;outside&quot; WebKit (e.g AppKit)

* Source/WebCore/page/scrolling/ScrollingTreeScrollingNode.cpp:
(WebCore::ScrollingTreeScrollingNode::scrollbarVisibilityDidChange):
(WebCore::ScrollingTreeScrollingNode::scrollbarMinimumThumbLengthDidChange):

Canonical link: <a href="https://commits.webkit.org/299829@main">https://commits.webkit.org/299829@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6d1afe6830b7d9be2f2fe9a82f4fdc83fa88b073

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/120342 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/40036 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/30688 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/126709 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/72415 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/0d052d10-b45c-41f1-b7f3-d2b2c3408d8f) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/122218 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/40733 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/48613 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91396 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/60688 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/510c7f32-a2a7-424f-9c67-8435f8236c32) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/123294 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32529 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/107882 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/71949 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/40390733-1f63-4236-b5f0-bd2afcbb1627) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/31562 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/25988 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/70329 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/102009 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/26172 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/129597 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/47263 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/35860 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/100012 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/47629 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/104066 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/99854 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25344 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45304 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23330 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/43908 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/47125 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/52830 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/46593 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/49939 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/48279 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebCore/page/scrolling/ScrollingTreeScrollingNode.cpp:
```diff
@@ -589,12 +589,14 @@ ScrollPropagationInfo ScrollingTreeScrollingNode::computeScrollPropagation(const
 void ScrollingTreeScrollingNode::scrollbarVisibilityDidChange(ScrollbarOrientation orientation, bool isVisible)
 {
-    scrollingTree()->scrollingTreeNodeScrollbarVisibilityDidChange(scrollingNodeID(), orientation, isVisible);
+    if (auto tree = scrollingTree())
```
Use `RefPtr` instead of `auto`.

tuankiet65 replied:
I changed it to `RefPtr`.

smfr commented on Source/WebCore/page/scrolling/ScrollingTreeScrollingNode.cpp:
```diff
@@ -589,12 +589,14 @@ ScrollPropagationInfo ScrollingTreeScrollingNode::computeScrollPropagation(const
 void ScrollingTreeScrollingNode::scrollbarMinimumThumbLengthDidChange(ScrollbarOrientation orientation, int minimumThumbLength)
 {
-    scrollingTree()->scrollingTreeNodeScrollbarMinimumThumbLengthDidChange(scrollingNodeID(), orientation, minimumThumbLength);
+    if (auto tree = scrollingTree())
```
Ditto.

tuankiet65 committed:
ScrollingTreeScrollingNode: null check scrollingTree() in some places
rdar://114615469
https://bugs.webkit.org/show_bug.cgi?id=298686

Reviewed by Simon Fraser.

AppKit might call into an orphan ScrollingTreeScrollingNode (node without
a scrolling tree) after the tree is torn down. As ScrollingTreeScrollingNode
holds a weak pointer to the owning scrolling tree, accessing the now-destroyed
tree crashes with a null pointer dereference. For now, null-check scrollingTree()
in places where the caller is from "outside" WebKit (e.g AppKit)

* Source/WebCore/page/scrolling/ScrollingTreeScrollingNode.cpp:
(WebCore::ScrollingTreeScrollingNode::scrollbarVisibilityDidChange):
(WebCore::ScrollingTreeScrollingNode::scrollbarMinimumThumbLengthDidChange):

Canonical link: https://commits.webkit.org/299829@main
